src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const { x = 0, y = 0 } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
 
-class DeliveryDrone {}
+    this.coords = {
+      x,
+      y,
+    };
+
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+     + `Chip version: ${this.chipVersion}, `
+     + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
```

Comment:
Same here with default value
```suggestion
  constructor(name, weight, coords, chipVersion) {
```